modin/pandas/groupby.py

```diff
@@ -987,6 +987,40 @@ def groupby_on_multiple_columns(df, *args, **kwargs):
     ],
 )
 class SeriesGroupBy(DataFrameGroupBy):
+    def __init__(self, df, *args, **kwargs):
+        if not isinstance(df, Series):
+            df = df.squeeze(axis=1)
+        ErrorMessage.catch_bugs_and_request_email(
+            not isinstance(df, Series),
+            f"SeriesGroupBy is supposed to obtain only Series objects, got: {type(df)}",
+        )
+        super(SeriesGroupBy, self).__init__(df.to_frame(), *args, **kwargs)
+        self._name = df.name or "__reduced__"
+
+    _NO_RENAME_METHODS = ["ngroup"]
+
+    def _build_series_wrapper(self, func):
+        def name_setter(*args, **kwargs):
+            if kwargs.get("axis", 0) == 1:
+                raise ValueError("No axis named 1 for object type Series")
```

Comment:
This check should happen in `Series.groupby` instead of `SeriesGroupBy` object.

Reply:
some `groupby` methods obtain `axis` argument, for example `shift`:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.DataFrameGroupBy.shift.html

Reply:
I see, but in this case if `axis` is not passed as a kwargs, but a positional arg it will not be checked, correct?

Reply:
Correct. It seems the only way to handle this is to overwrite all methods that have `axis` arguments in `SeriesGroupby` object just to process this check, do we actually need it?

Reply:
or we can discard [this](https://github.com/modin-project/modin/pull/2812#pullrequestreview-604115268) approach and return to the previous one, however it raises a more complicated question ([link](https://github.com/modin-project/modin/pull/2812#pullrequestreview-604115268))

Reply:
@devin-petersohn so what is an appropriate way in your opinion?

Reply:
Can we check axis arguments in `DataFrameGroupBy` in the same way we do in the `dataframe.py` and `series.py` files with a `self._get_axis_number`?

Reply:
it's not necessary if we continue to use Series at the base class [[comment]](https://github.com/modin-project/modin/pull/2812#issuecomment-802044851)